### ISSUE TITLE
Using Ruby Gems and Licensing

### ISSUE THREAD

**Issue by ThomasBush** [WestMichiganRubyTraining/discussion#3]
I am new to the Ruby community, but I remember Bill saying it was about sharing with each other, open source software for the most part, etc.  With that in mind - I had a "GR Web Dev" Meetup meeting on Monday.  The topic of this meetup was open source software, licensing and liability.  I quickly panicked as I realized I am trying to build an application that uses ruby gems.  

I have noticed an MIT license for example on Twitter Bootstrap gem, which tells me I am free to use, but have noticed some gems don't have a license at all.  This worries me a bit.  Is anyone experienced here?  How do I handle a gem with no license? Any general information on this topic?

**Comment by billgathen** [comment]
I'm surprised that didn't come up as a topic at the meeting. Plenty of gems don't have licenses because the authors don't care who uses their gem (or don't expect anyone to). I believe I put an MIT license on my yieldmanager gem because it was the most-lenient.

I'd be interested in what they had to say at the meeting about liability. I've never heard of anyone being the victim of legal action because of using a gem that's freely-available for download on the most popular dev repository on the net. You'd think if they wanted to restrict its use, they'd host it themselves.

That said, I believe Zach Holman (from GitHub) said recently he wished someone would write a spider that would crawl GH and send a pull request containing the MIT license to any public repo that didn't have a license. :-)

The RubyRogues did an episode awhile back about open source licensing (http://rubyrogues.com/075-rr-open-source-licenses/). I haven't listened to it, but you might give it a listen and let us know what you find out.

**Comment by ThomasBush** [comment]
Thanks for the info Bill.  The talk was not ruby centered so gems were not discussed, more general and I didn't think to ask until it was too late.  My concern would be creating something and than trying to sell advertising for example on it.  Then being forced to open source every bit of code because I use a gem that contained a GPL v2 license which would force me to perpetuate this license if I wanted to use the gem.  

Basically, two lawyers gave presentations as part of this meeting that were some what alarming for me so I figured I would ask how these concepts would apply to the ruby community - in particular ruby gems.
